### PR TITLE
make the linter config attractive

### PR DESCRIPTION
**Pull request by faddat**
Closes: #XXX

## What is the purpose of the change

Long ago, I used .golangci.yml in Osmosis to learn how to fully configure golangci-lint. 

THis PR aims to fix that, while masking changes that Dev mentioned were unnecessary. 

## Brief Changelog

* Make .golangci.yml quality
* change the Makefile so that there's no longer a need to separately invoke gofumpt

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no) no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no) no
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented) not applicable

### CODE REVIEW

File: osmomath/sigfig_round.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ func SigFigRound(d sdk.Dec, tenToSigFig sdk.Int) sdk.Dec {
 	// take note of floor div, vs normal div
 	k := uint64(0)
 	dTimesK := d
-	for ; dTimesK.LT(pointOne); k += 1 {
+	for ; dTimesK.LT(pointOne); k++ {
```
Comment (ValarDragon) [inline osmomath/sigfig_round.go:18]: which linter said this, can we turn it off

File: .golangci.yml [hunk 2]
```diff
@@ -10,106 +10,74 @@ linters:
...
     - goconst
-    # - gocritic <- re enable later to catch capitalized local variables
-    #    - gocyclo <- disabled because it won't work with how we cosmos
-    # - godot
-    #    - godox <- detects TODO/BUG/FIXME and we may wantnthis later but isn't appropriate now
-    #    - goerr113 <- disabled due to lack of comprehension
-    - gofmt
-    # - gofumpt <- disabled https://github.com/osmosis-labs/osmosis/issues/1309
+    - gocritic # <- re enable later to catch capitalized local variables
+    - gofumpt
```
Comment (ValarDragon) [inline .golangci.yml:21]: Lets leave this deleted, it doesn't have good IDE support.

File: .golangci.yml [hunk 3]
```diff
@@ -10,106 +10,74 @@ linters:
...
     - goheader
-    - goimports
-    #    - gomoddirectives <- disables replaces
     - gomodguard
     - goprintffuncname
-#    - gosec <- triggers too much for this round and should be re-enabled later
+    - gosec # <- triggers too much for this round and should be re-enabled later
```
Comment (ValarDragon) [inline .golangci.yml:25]: Can this be changed in a separate PR?

File: .golangci.yml [hunk 4]
```diff
@@ -10,106 +10,74 @@ linters:
...
     - makezero
     - misspell
     - nakedret
-#    - nestif <- ?
-    # - nilerr <- needs go 1.18 support
     - nilnil
-    #    - nlreturn <- disabled as it doesn't auto-fix something simple and doesn't add anything useful
-    # - noctx <- needs go 1.18 support
-    #    - nolintlint <- disabled because we use nolint in some places
+    - nolintlint # <- disabled because we use nolint in some places
```
Comment (ValarDragon) [inline .golangci.yml:34]: Lets delete this?

Nolints are perfectly fine to have in the codebase.

Comment (faddat) [reply .golangci.yml:34]: nolintlint just forces us to 

//nolint:whatever per-line, I'd prefer to keep it ? 

Basically it is an enforcer of //nolint:whatever hygene.

File: .golangci.yml (hunk 3 above)
Comment (faddat) [reply .golangci.yml:25]: yes.

File: osmomath/sigfig_round.go (hunk 1 above)
Comment (faddat) [reply osmomath/sigfig_round.go:18]: totally, it's revive's increment-decrement, and can get rid of it :)

File: .golangci.yml (hunk 2 above)
Comment (faddat) [reply .golangci.yml:21]: with gofumpt, what I have been doing, is using it to replace both goimports and gofmt

if your ide supports golangci-lint, it will now gofumpt automagically.

**Comment by faddat** [comment]
After talking to Adam on slack, I think that this pr should be kept super minimal, and expanded after the concentrated liquidity branch is merged.

**Comment by faddat** [comment]
recreating such that it doesn't change code at this time.

**Comment by faddat** [comment]
Hey, you asked about the wierd error?  it has returned:

```
golangci-lint run ./... --fix
WARN [runner] Can't run linter goanalysis_metalinter: gofmt: can't extract issues from gofmt diff output "--- /Users/faddat/osmosis/x/twap/client/grpc/grpc_query.go.orig\t2022-11-30 01:04:06\n+++ /Users/faddat/osmosis/x/twap/client/grpc/grpc_query.go\t2022-11-30 01:04:06\n@@ -1,4 +1,4 @@\n-package grpc \n+package grpc\n \n // THIS FILE IS GENERATED CODE, DO NOT EDIT\n // SOURCE AT `proto/osmosis/twap/v1beta1/query.yml`\n@@ -49,4 +49,3 @@\n \tctx := sdk.UnwrapSDKContext(grpcCtx)\n \treturn q.Q.ArithmeticTwap(ctx, *req)\n }\n-\n": can't parse patch: parsing time "2022-11-30 01:04:06" as "2006-01-02 15:04:05 -0700": cannot parse "" as "-0700" 
ERRO Running error: 1 error occurred:
        * can't run linter goanalysis_metalinter: gofmt: can't extract issues from gofmt diff output "--- /Users/faddat/osmosis/x/twap/client/grpc/grpc_query.go.orig\t2022-11-30 01:04:06\n+++ /Users/faddat/osmosis/x/twap/client/grpc/grpc_query.go\t2022-11-30 01:04:06\n@@ -1,4 +1,4 @@\n-package grpc \n+package grpc\n \n // THIS FILE IS GENERATED CODE, DO NOT EDIT\n // SOURCE AT `proto/osmosis/twap/v1beta1/query.yml`\n@@ -49,4 +49,3 @@\n \tctx := sdk.UnwrapSDKContext(grpcCtx)\n \treturn q.Q.ArithmeticTwap(ctx, *req)\n }\n-\n": can't parse patch: parsing time "2022-11-30 01:04:06" as "2006-01-02 15:04:05 -0700": cannot parse "" as "-0700"
```

**Comment by faddat** [comment]
```
golangci-lint run ./... --fix         
WARN [runner] Can't run linter goanalysis_metalinter: goimports: can't extract issues from gofmt diff output "--- /Users/faddat/osmosis/x/epochs/types/genesis.pb.go.orig\t2022-11-30 01:26:25\n+++ /Users/faddat/osmosis/x/epochs/types/genesis.pb.go\t2022-11-30 01:26:25\n@@ -5,14 +5,15 @@\n \n import (\n \tfmt \"fmt\"\n-\t_ \"github.com/gogo/protobuf/gogoproto\"\n-\tproto \"github.com/gogo/protobuf/proto\"\n-\t_ \"github.com/gogo/protobuf/types\"\n-\tgithub_com_gogo_protobuf_types \"github.com/gogo/protobuf/types\"\n \tio \"io\"\n \tmath \"math\"\n \tmath_bits \"math/bits\"\n \ttime \"time\"\n+\n+\t_ \"github.com/gogo/protobuf/gogoproto\"\n+\tproto \"github.com/gogo/protobuf/proto\"\n+\t_ \"github.com/gogo/protobuf/types\"\n+\tgithub_com_gogo_protobuf_types \"github.com/gogo/protobuf/types\"\n )\n \n // Reference imports to suppress errors if they are not otherwise used.\n": can't parse patch: parsing time "2022-11-30 01:26:25" as "2006-01-02 15:04:05 -0700": cannot parse "" as "-0700" 
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: goimports: can't extract issues from gofmt diff output "--- /Users/faddat/osmosis/x/epochs/types/genesis.pb.go.orig\t2022-11-30 01:26:25\n+++ /Users/faddat/osmosis/x/epochs/types/genesis.pb.go\t2022-11-30 01:26:25\n@@ -5,14 +5,15 @@\n \n import (\n \tfmt \"fmt\"\n-\t_ \"github.com/gogo/protobuf/gogoproto\"\n-\tproto \"github.com/gogo/protobuf/proto\"\n-\t_ \"github.com/gogo/protobuf/types\"\n-\tgithub_com_gogo_protobuf_types \"github.com/gogo/protobuf/types\"\n \tio \"io\"\n \tmath \"math\"\n \tmath_bits \"math/bits\"\n \ttime \"time\"\n+\n+\t_ \"github.com/gogo/protobuf/gogoproto\"\n+\tproto \"github.com/gogo/protobuf/proto\"\n+\t_ \"github.com/gogo/protobuf/types\"\n+\tgithub_com_gogo_protobuf_types \"github.com/gogo/protobuf/types\"\n )\n \n // Reference imports to suppress errors if they are not otherwise used.\n": can't parse patch: parsing time "2022-11-30 01:26:25" as "2006-01-02 15:04:05 -0700": cannot parse "" as "-0700"

```

**Comment by faddat** [comment]
summary:

* Makefile and .golangci.yml changed
* goimports and gofmt dropped in favor of gofumpt, due to the errors above
* no other changes in terms of which linters we use

File: .golangci.yml (hunk 4 above)
Comment (ValarDragon) [reply .golangci.yml]: oh gothca

File: .golangci.yml (hunk 2 above)
Comment (ValarDragon) [reply .golangci.yml:20]: Looking at the lints from gofumpt, I really don't think this is a win to have as overhead on every PR/commit

**Comment by ValarDragon** [comment]
We have to fix the lint failures on this PR (or comment out those linters and make a new GH issue for them)

**Comment by faddat** [comment]
totally!  sorry I missed that :P

**Review by ValarDragon — APPROVED** [review]
